convert2rhel/systeminfo.py

```diff
@@ -397,10 +413,58 @@ def corresponds_to_rhel_eus_release(self):
         For example if we detect CentOS Linux 8.4, the corresponding RHEL 8.4 is an EUS release, but if we detect
         CentOS Linux 8.5, the corresponding RHEL 8.5 is not an EUS release.
 
-        :return: Whether or not the current system has a EUS correspondent in RHEL.
+        :return: Whether or not the current system has an EUS correspondent in RHEL.
         :rtype: bool
         """
-        return "%s.%s" % (self.version.major, self.version.minor) in EUS_MINOR_VERSIONS
+        eus_release_date = EUS_MINOR_VERSIONS.get("%s.%s" % (self.version.major, self.version.minor), False)
+        if eus_release_date:
+            eus_release_date = dt.strptime(eus_release_date, "%Y-%m-%d")
+            current_datetime = dt.today()
+            if not self.has_internet_access:
+                return self.datetime_comparison_returns(current_datetime, eus_release_date)
+            elif current_datetime > eus_release_date:
+                try:
+                    latest_release_version = self.get_latest_distro_release_version()
+                except RepoqueryFailure:
+                    return self.datetime_comparison_returns(current_datetime, eus_release_date)
+
+                latest_major, latest_minor = latest_release_version.split(".")
+                if int(latest_minor) > int(self.version.minor):
+                    return "%s.%s" % (self.version.major, self.version.minor) in EUS_MINOR_VERSIONS
+                if int(latest_minor) < int(self.version.minor):
+                    pass
+                    # corner case that should not happen, need to decide what to do in case we land in this condition
+                    # where the users system on a version that hasn't been released yet
+        return False
+
+    def datetime_comparison_returns(self, current_datetime, eus_release_date):
+        if current_datetime > eus_release_date:
+            return "%s.%s" % (self.version.major, self.version.minor) in EUS_MINOR_VERSIONS
+        return False
+
+    def get_latest_distro_release_version(self):
+        """Extract minor version from release name using yum list"""
+        cmd = [
+            "repoquery",
+            "--setopt=exclude=",
+            "--quiet",
+            "--qf",
+            "C2R\\t%{VERSION}",
+            self.release_pkg,
+        ]
+        yum_output, return_code = run_subprocess(cmd, print_output=False)
+        if return_code != 0:
+            self.logger.debug("Got the following output: %s", yum_output)
+            raise RepoqueryFailure("Repoquery failed to execute")
```

Comment:
Since we won't be using the message here, is there any purpose in keeping it? We could transform this into a comment instead, I think... Would be more valuable to know what are the possibly failure for repoquery in this case instead of assigning a message to this exception and not using it anywhere else.

Reply:
Okay so remove the exception and keep the debug?

Reply:
I think you can just `raise RepoqueryFailure` instead and not pass any message to it.

Reply:
> I would always pass a message with an Exception. The reason is that the code that catches the exception is in charge of making the decisions. So the code which raises the exception doesn't know whether a message is needed or not.

For that, then, let's make the message more meaningful and provide some context of what exactly failed with the repoquery

Reply:
Okay so I can just add the debug info into the exception